components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -25,333 +24,155 @@ import org.mozilla.appservices.logins.GleanMetrics.LoginsStore as LoginsStoreMet
 import mozilla.components.service.glean.private.CounterMetricType
 import mozilla.components.service.glean.private.LabeledMetricType
 
+// uniffi has unfortunate names for errors: https://github.com/mozilla/uniffi-rs/issues/442
+typealias LoginsStorageException = LoginsStorageErrorException
+
 /**
- * LoginsStorage implementation backed by a database.
+ * An artifact of the uniffi conversion - a thin-ish wrapper around a
+   PasswordStore.
```

Comment:
What does the wrapper add that we couldn't implement in rust/uniffi? It's probably worth recording as part of the comment here, so that our future selves have the context if uniffi gains more capabilities in future.

Reply:
yes, good call. IIRC, there are 2 main things:

* metrics :/
* The lock/unlock semantics - when we "unlock" we actually create a new `PasswordStore` and when we lock it, we drop it. However, the `DatabaseLoginsStorage` stays alive.

So I think we can drop the former once we get a glean API in rust and the latter once we drop sqlcipher.